cdci_data_analysis/analysis/queries.py

```diff
@@ -257,25 +245,26 @@ def print_form_dictionary_list(self,l):
         else:
             return l
 
-
-
     def get_parameters_list_as_json(self,prod_dict=None):
         l=[ {'query_name':self.name}]
 
         for par in self._parameters_list:
             l.append(par.reprJSON())
 
-
         return json.dumps(l)
 
+    # Check if the given query cn be executed given a list of roles extracted from the token
+    def check_query_roles(self, roles, par_dic):
+        return True, []
```

Comment:
what's the second argument? In general, I think it is a bit confusing when tuples are returned without annotation. I like to use dict in a case like this with informtive keys.
Or type annotations (with typing module).  But in this case it is indeed not clear  for me what it means without reading more.
but maybe just a comment would do, for us

Reply:
Since we want to show the list of the required role(s) for a given query in case this is unauthorized (ie [](https://github.com/oda-hub/dispatcher-app/blob/2be8a88c68a621ba1caf462aeb070054c80d2fb0/cdci_data_analysis/analysis/instrument.py#L254)), I return this directly from the method that is supposed to check for it (eg [another example](https://github.com/oda-hub/dispatcher-app/blob/2be8a88c68a621ba1caf462aeb070054c80d2fb0/cdci_data_analysis/plugins/dummy_instrument/data_server_dispatcher.py#L94)).

But I agree that actually, a dictionary is more informative.

Reply:
Indeed, makes sense, and I think it was even me who asked for this functionality!
But I did not remember.

Reply:
It returns now a dictionary, easier to handle and less error-prone